pySDC/implementations/controller_classes/controller_nonMPI.py

```diff
@@ -799,13 +820,33 @@ def adaptivity(self, MS):
             h_opt = L.params.dt * 0.9 * (L.params.e_tol / L.status.error_embedded_estimate)**(1. / order)
 
             # distribute step sizes
-            if len(MS) > 1:
-                raise NotImplementedError('Adaptivity only implemented for 1 rank just yet')
-            else:
-                L.params.dt = h_opt
+            L.status.dt_new = h_opt
 
             # check whether to move on or restart
             if L.status.error_embedded_estimate >= L.params.e_tol:
                 self.restart[i] = True
-            else:
-                self.restart[i] = False
+                S.status.restart = True
+
+    def adaptivity_update_step_sizes(self, active_slots):
```

Comment:
Do we agree that having all those function as part of the controller makes the controller even worse in terms of readability and maintainability?